code/vocabulary.py

```diff
@@ -159,3 +161,54 @@ def replace_words(words: Sequence[str]) -> Sequence[str]:
             # fall back to default implementation for error-robustness
             logging.error("phrase replacer failed!")
             return actions.next(words)
+
+def _create_vocabulary_entries(spoken_form, written_form, type):
+    """Expands the provided spoken form and written form into multiple variants based on
+    the provided type, which can be either "name" to add a possessive variant or "noun"
+    to add plural.
+    """
+    entries = {spoken_form: written_form}
+    if type == "name":
+        entries[f"{spoken_form}s"] = f"{written_form}'s"
+    elif type == "noun":
+        entries[f"{spoken_form}s"] = f"{written_form}s"
+    return entries
+
+# See https://github.com/wolfmanstout/talon-vocabulary-editor for an experimental version
+# of this which tests if the default spoken form can be used instead of the provided phrase.
+def _add_selection_to_csv(
+    phrase: Union[Phrase, str], type: str, csv: str, headers: Tuple[str, str]
+):
+    written_form = actions.edit.selected_text().strip()
+    if phrase:
+        spoken_form = " ".join(actions.dictate.parse_words(phrase))
+    else:
+        is_acronym = re.fullmatch(r"[A-Z]+", written_form)
+        spoken_form = " ".join(written_form) if is_acronym else written_form
+    phrases = get_list_from_csv(csv, headers=headers, write_default=False)
```

Comment:
I think this is wrong, because get_list_from_csv causes the file to be watched and auto-reloads when it changes, whereas here you want a one-time read of the csv file.

Reply:
I just reached out to Ryan about this, because it seems undesirable that Talon would do this when resource.open is invoked from a command resolution (vs. file initialization). In a nutshell, he didn't disagree, but he wants to get away from resource.open entirely so his advice here was to work around this by simply reusing the earlier result of get_list_from_csv, but without going through ctx to avoid issues with changing the file. That is now implemented here. I also reverted my changes to get_list_from_csv which are no longer needed.